lib/chat/rooms/room.ex

```diff
@@ -106,22 +106,30 @@ defmodule Chat.Rooms.Room do
 
   def hash(pub_key), do: Base.encode16(pub_key, case: :lower)
 
-  def decrypt_identity(ciphered, %Identity{} = me, room_public_key) do
-    secret = Enigma.compute_secret(me.private_key, room_public_key)
+  def encrypt_identity(%Identity{} = room_identity, secret) do
+    room_identity
+    |> Identity.to_strings()
+    |> Enum.at(1)
+    |> Enigma.cipher(secret)
+  end
 
-    ciphered
+  def decrypt_identity(encrypted, secret) do
+    encrypted
     |> Enigma.decipher(secret)
     |> then(&["", &1])
     |> Identity.from_strings()
   end
 
-  defp encrypt_identity(room_identity, user_public_key) do
+  def decrypt_identity_with_key(encrypted, %Identity{} = me, room_public_key) do
```

Comment:
Correct. Now renamed `encrypt_identity/2` -> `cipher_identity/2`.
So they have the same arity, but different internal logic for a little and the same type of arguments on the enter. So that's the reason